pkg/services/govmomi/clustermodules/provider.go

```diff
@@ -88,20 +91,19 @@ func (cm *provider) DoesModuleExist(ctx context.Context, moduleUUID string, clus
 		return false, nil
 	}
 
-	modules, err := cm.manager.ListModules(ctx)
-	if err != nil {
-		return false, err
+	_, err := cm.manager.ListModuleMembers(ctx, moduleUUID)
+	if err == nil {
+		log.V(4).Info("Cluster module exists", "moduleUUID", moduleUUID)
+		return true, nil
 	}
 
-	for _, mod := range modules {
-		if mod.Cluster == clusterRef.Value && mod.Module == moduleUUID {
-			log.V(4).Info("Cluster module does exist", "computeClusterRef", clusterRef)
-			return true, nil
-		}
+	// TODO: Once govmomi supports error checking for rest client, replace the below function
+	if strings.Contains(err.Error(), fmt.Sprintf("%d %s", http.StatusNotFound, http.StatusText(http.StatusNotFound))) {
```

Comment:
When you bump govmomi and replace this. Can you please check if we can also get rid of `util.IsNotFoundError` above? (just randomly noticed it, wasn't aware of that util before)

(please check if we can get rid of both usages of that util and the util itself)

Reply:
done